detektive/src/main/kotlin/cash/detektive/javacompat/AnnotatePublicApisWithJvmOverloads.kt

```diff
@@ -50,14 +56,24 @@ class AnnotatePublicApisWithJvmOverloads(config: Config) : Rule(config) {
     }
     if (isApplicable(element, bindingContext[BindingContext.DECLARATION_TO_DESCRIPTOR, element])) {
       if (!element.annotationEntries.any { it.isOfType(JvmOverloads::class) }) {
-        report(
-          CodeSmell(
-            issue,
-            Entity.atName(element),
-            "Public ${elementType.name.lowercase()} '${element.nameAsSafeName}' " +
-              "with default arguments, but without @JvmOverloads annotation"
-          )
-        )
+        val message = "Public ${elementType.name.lowercase()} '${element.nameAsSafeName}' " +
+          "with default arguments, but without @JvmOverloads annotation"
+
+        if (autoCorrect) {
+          if (!(element as KtElement).isSuppressedBy(ruleId, aliases, ruleSetConfig.parentPath)) {
```

Comment:
Ignoring autoCorrect in cases where this rule is suppressed for the constructor or function.